hedera-cryptography/hedera-cryptography-tss/src/main/java/com/hedera/cryptography/tss/api/TssService.java

```diff
@@ -0,0 +1,167 @@
+     *
+     * @param participantDirectory the pending participant directory that we should generate the private share for
+     * @param validTssMessages the TSS messages to extract the private shares from. They must be previously validated.
+     * @return a sorted by sharedId list of private shares the current participant owns, or null if there aren't enough shares to meet the threshold.
+     */
+    @Nullable
+    List<TssPrivateShare> decryptPrivateShares(
+            @NonNull TssParticipantDirectory participantDirectory, @NonNull List<TssMessage> validTssMessages);
+
+    /**
+     * Aggregate a threshold number of {@link TssPrivateShare}s.
+     * It is the responsibility of the caller to ensure that the list of private shares meets the required threshold.
+     * If the threshold is not met, the private key returned by this method will be invalid.
+     *
+     * @param privateShares the private shares to aggregate
+     * @return the aggregate private key
+     */
+    @NonNull
+    PairingPrivateKey aggregatePrivateShares(@NonNull List<TssPrivateShare> privateShares);
+
+    /**
+     * Compute all public shares for all the participants in the scheme.
+     *
+     * @param participantDirectory the pending participant directory that we should generate the message for
+     * @param validTssMessages the TSS messages to extract the private shares from. They must be previously validated.
+     * @return  a sorted by the sharedId list of public shares, or null if there aren't enough messages to meet the threshold
+     */
+    @Nullable
+    List<TssPublicShare> computePublicShares(
+            @NonNull TssParticipantDirectory participantDirectory, @NonNull List<TssMessage> validTssMessages);
+
+    /**
+     * Aggregate a threshold number of {@link TssPublicShare}s.
+     * It is the responsibility of the caller to ensure that the list of public shares meets the required threshold.
+     * If the threshold is not met, the public key returned by this method will be invalid.
+     * This method is used for two distinct purposes:
+     * <ul>
+     *     <li>Aggregating public shares to produce the Ledger ID</li>
+     *     <li>Aggregating public shares derived from all commitments, to produce the public key for a given share</li>
+     * </ul>
+     *
+     * @param publicShares the public shares to aggregate
+     * @return the interpolated public key
+     */
+    @NonNull
+    PairingPublicKey aggregatePublicShares(@NonNull List<TssPublicShare> publicShares);
+
+    /**
+     * Sign a message using the private share's key.
+     * @param privateShare the private share to sign the message with
+     * @param message the message to sign
+     * @return the signature, which will be in the group opposite to the group of the public key
+     */
+    @NonNull
+    TssShareSignature sign(@NonNull TssPrivateShare privateShare, @NonNull byte[] message);
```

Comment:
does the sign method really belong in this class? It seems it should be the responsibility of the `privateShare`

Reply:
All the behavior in the library objects has been moved to the service class to make them easily convertible to protobuf and to locate all library operations in a central place.
The object-oriented approach still applies to the pairings API.